### DISCUSSION TITLE
GitHub badges

### DISCUSSION THREAD

**Discussion by ninadicara** [very-good-science/data-hazards#93] [Ideas]
People are interested in having some kind of Data Hazards badge on a repository.

It could link to your data hazards self assessment. Could also show if it is out of date after a major release.

**Comment by NatalieZelenka** [comment]
Could also require a certain number of assessments with a certain user base?

**Comment by jqdq** [comment]
After your workshop, I did some prototyping in JavaScript. I've managed to create _something_, but it only works on GH Pages and classic webpages. This is because GitHub disables scripts in README files. Let me know what you think!
https://pogromcapapai.github.io/dh-badge/

**Reply by NatalieZelenka** [reply]
Wow thanks so much Jakub this is the coolest! We are working on a section of our website to help people display the hazards. I will work on trying to include this at our next co-working call!

**Reply by ninadicara** [reply]
No way - this is amazing!! Thank you so much @PogromcaPapai !
Looking at your example is also a reminder to me that the outlines on some of the Hazards are different and this makes some look a bit out of place when put together, but this should be fixed by #118 in a few months time when they're done

**Reply by jqdq** [reply]
No problem :D I really love this project and will gladly help in anything programming-related.
I think I know how to make it usable in README files and I'm going to try implementing it today. I will keep you updated!
I will probably also think about how to retrieve the labels from a self-assessment file. Informing about outdated assessments should also be possible, but this could be another badge?
If you have any requests or ideas, please do @ me.

**Reply by jqdq** [reply]
[![Example labels](https://notch-marble-primrose.glitch.me/labels?d=general,misuse,automates,privacy,bias,classifies,consent,difficult,direct,environment,ignores,lacks)](https://github.com/very-good-science/data-hazards)

```markdown
[![Example labels](https://notch-marble-primrose.glitch.me/labels?d=general,misuse,automates,privacy,bias,classifies,consent,difficult,direct,environment,ignores,lacks)](https://github.com/very-good-science/data-hazards)
```
It's alive!!

**Reply by ninadicara** [reply]
This is so cool Jakub - thank you!

I think the next step is to work out how to incorporate your work into this repo, and add the instructions on how to use it to a page in the 'How to Use' section (even if we improve the functionality further later, I think just being able to display them would be great!)

I've started by [adding a page](https://github.com/very-good-science/data-hazards/blob/main/site/contents/materials/presenting.md) ([you can see it on the website here](https://very-good-science.github.io/data-hazards/contents/materials/presenting.html)) about how to present the Hazards, and then set up an empty sub-page which would contain documentation on how to use this tool in the `presenting` folder there. 

I'm not sure if it is easy (or possible!) to transfer what's happening on glitch to here since this only serves static pages?

**Reply by jqdq** [reply]
Yeah, I've tried deploying it using static GH Pages, but GitHub disallows using JavaScript in README files (as far as I know). Because of this, you can't generate the badge dynamically in the browser (which was why I couldn't include it in README files, but there was no such problem in GH Pages). Because of that, I've decided to host it on glitch. I've seen people use vercel for it, and I think vercel only allows client-side code execution, so maybe it is possible to make these on the client's side.

I will add the documentation here in some time. Should I create a new branch for this?

Also, I've seen the app from pull request #101, and I think it might be useful to automatically generate a badge with the selected hazard labels. What do you think?
